app/src/main/java/com/android/feedme/model/viewmodel/InputViewModel.kt

```diff
@@ -35,15 +34,18 @@ class InputViewModel(val context: Context? = null) : ViewModel() {
   val listOfIngredientMetadatas: StateFlow<List<IngredientMetaData?>> = _listOfIngredientMetadatas
 
   // StateFlow to hold the total number of complete ingredients
-  private val _totalCompleteIngredientMetadatas = MutableStateFlow(0)
-  val totalCompleteIngredientMetadatas: StateFlow<Int> = _totalCompleteIngredientMetadatas
+  private val _isComplete =
+      MutableStateFlow(
+          _listOfIngredientMetadatas.value.count { isCompleteIngredient(it) } ==
+              _totalIngredientEntriesDisplayed.value - 1)
```

Comment:
I don't fully understand what you mean by _totalIngredientEntriesDisplayed.value -1 ?

Reply:
I think the `-1` is there because our list is expected to have at least one `null` element, therefore we check if our count of complete ingredients in the list matches the number of ingredients to display minus the `null` element. correct me if i'm wrong @FacerOfGod !

Reply:
That is correct :)